### PR TITLE
Fix bug in Rmdir; Rewrite unit tests; Introduce Travis CI

### PR DESCRIPTION
**Pull request by cxwangyi**
Fixes https://github.com/wangkuiyi/etcd/issues/3
Fixes https://github.com/wangkuiyi/etcd/issues/5

### CODE REVIEW

File: etcd_test.go [hunk 1]
```diff
@@ -18,47 +22,36 @@ func ExampleNew() {
...
-		return
-	}
+	assert := assert.New(t)
+
+	c, e := New("http://localhost:4001,http://localhost:2379")
+	assert.Nil(e)
+
+	c.Rmdir("/home/yi")
+
+	assert.Nil(c.Mkdir("/home/yi"))
```
Comment (chenkaitopic) [inline etcd_test.go]: Let's add another line assert.NotNil(c.Mkdir("/home/yi")) to verify Mkdir will return error if dir exists.

**Comment by chenkaitopic** [comment]
We don't have a "Delete" method that only deletes a particular key-value pair. Should we add it?

File: etcd_test.go (hunk 1 above)
Comment (cxwangyi) [inline etcd_test.go]: Done

**Comment by chenkaitopic** [comment]
LGTM

**Pull request merged**